src/nix/app.cc

```diff
@@ -85,9 +85,9 @@ UnresolvedApp Installable::toApp(EvalState & state)
 
     else if (type == "derivation") {
         auto drvPath = cursor->forceDerivation();
-        auto outPath = cursor->getAttr(state.sOutPath)->getString();
-        auto outputName = cursor->getAttr(state.sOutputName)->getString();
-        auto name = cursor->getAttr(state.sName)->getString();
+        auto outPath = cursor->getAttr("outPath")->getString();
+        auto outputName = cursor->getAttr("outputName")->getString();
+        auto name = cursor->getAttr("name")->getString();
```

Comment:
I don't understand this and similar changes. Not that this is on the hot path, but isn't it a worse for performance to replace a lookup of a symbol by a lookup of a string?

Reply:
we changed these for readability. ioo it was preferrable to do it like this instead of as
```
auto name = cursor->getAttr(state.symbols[state.sName])->getString();
```
but ultimately both of them are totally valid.